docs/modules/ROOT/pages/how-tos/configuring/customizing-the-build.adoc

```diff
@@ -1,65 +1,133 @@
 
 
+== Changing parameters
+
 .Procedure
 
-To customize the build pipeline in any way, start by taking the following steps:
+. In the PipelineRun YAML files in the `.tekton` directory, customize the parameters in the `.spec.params` section:
+.. You can change the default `value` of existing parameters. For example, for PRs, you could change the value of `image-expires-after` from the default `5d` to `1w`, so images built for PRs expire after a week rather than 5 days.
++
+[source,diff]
+----
+ spec:
+   params:
+   - name: image-expires-after
+-    value: 5d
++    value: 1w
+   - name: dockerfile
+     value: Dockerfile
+----
 
-. In your preferred IDE, navigate to the `.tekton` directory in the repository of your component.
+.. You can also add parameters. New parameters must include a `name` and `value`.
++
+[source,diff]
+----
+ spec:
+   params:
++  - name: hermetic
++    value: "true"
+   - name: image-expires-after
+     value: 1w
+   - name: dockerfile
+     value: Dockerfile
+----
++
+Refer to the `params` section in your Pipeline to see the available parameters.
+When looking at a PipelineRun file, you can typically find the Pipeline definition in:
++
+* The PipelineRun file itself, inlined in `.spec.pipelineSpec`
+* A separate file in the `.tekton` directory, referenced by name in `.spec.pipelineRef`
+* An external location, referenced by a resolver in `.spec.pipelineRef`
 
-. Open the relevant YAML files:
-.. The file ending in `-pull-request.yaml` determines how {ProductName} responds when you open pull requests.
-.. The file ending in `-on-push.yaml` determines how {ProductName} responds when you push commits.
+. Commit your changes to the repository of the component.
 
-== Changing parameters
+
+== Configuring timeouts
+
+By default, your pipeline has a timeout of 1 hour (as does each task in the pipeline individually).
+Your builds may take longer than that, in which case you will need to increase the timeouts accordingly.
 
 .Procedure
 
-. In the relevant YAML files, in the `.tekton` directory, customize the parameters of the build pipeline by changing the pipeline `params`:
-.. To make sure your changes are not overriden, change params with the paths `.spec.params`.
-.. You can change the default `value` of existing parameters. For example, for PRs, you could set a value for `image-expires-after`, like `2w` or `5d`, so images built for PRs do not last indefinitely.
-.. You can also add parameters. New parameters must include a `name` and `value`.
-. Commit your changes to the repository of the component.
+For example, let's configure the timeouts for a pipeline where the build itself may take up to 3 hours
+and antivirus scanning may take up to 2 hours. We'll set corresponding timeouts for both the tasks
+and a 6 hour total timeout for the pipeline.
```

Comment:
I know that https://tekton.dev/docs/pipelines/pipelineruns/#configuring-a-failure-timeout includes the fact that the pipeline timeout should be the sum of the tasks + finally timeouts, but should we call this out here?

Reply:
If I understand the Tekton docs correctly, this is not something the users need to care about

> All three sub-fields are optional, and will be automatically processed according to the following constraint:
>
> *   timeouts.pipeline >= timeouts.tasks + timeouts.finally

If they set `tasks` and/or `finally` and don't set `pipeline`, IIUC Tekton will automatically set it to the sum of `tasks` and `finally`

If they set only `pipeline`, then the `tasks` and `finally` sections don't have their own timeouts, only the pipeline as a whole has a timeout